scan.hcl

```diff
@@ -0,0 +1,22 @@
+repository {
+  go_modules = true
+  osv = true
+  secrets {
+    all = true
+  } 
+  dependabot {
+    required = true
+    check_config = true
+  }
+  
+  plugin "semgrep" {
```

Comment:
I love semgrep, will this allow us to add custom rules as well?

Reply:
Yup, can add any custom ruleset here, https://github.com/hashicorp/boundary/pull/2945/files#diff-5c4fa294b508f98a963471292bd581cab3912953dc491e452488a6a7e1bf4dbaR15.